sparse/unit_test/Test_Sparse_csr2csc.hpp

```diff
@@ -0,0 +1,182 @@
+// met:
+//
+// 1. Redistributions of source code must retain the above copyright
+// notice, this list of conditions and the following disclaimer.
+//
+// 2. Redistributions in binary form must reproduce the above copyright
+// notice, this list of conditions and the following disclaimer in the
+// documentation and/or other materials provided with the distribution.
+//
+// 3. Neither the name of the Corporation nor the names of the
+// contributors may be used to endorse or promote products derived from
+// this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY NTESS "AS IS" AND ANY
+// EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+// PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL NTESS OR THE
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+// PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+// LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+// NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+// SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// Questions? Contact Siva Rajamanickam (srajama@sandia.gov)
+//
+// ************************************************************************
+//@HEADER
+*/
+
+#include "KokkosSparse_csr2csc.hpp"
+#include "KokkosKernels_TestUtils.hpp"
+
+namespace Test {
+template <class ScalarType, class LayoutType, class ExeSpaceType>
+void doCsr2Csc(size_t m, size_t n, ScalarType min_val, ScalarType max_val,
+               bool fully_sparse = false) {
+  RandCsMatrix<ScalarType, LayoutType, ExeSpaceType> csrMat(
+      m, n, min_val, max_val, fully_sparse);
+
+  auto cscMat = KokkosSparse::csr2csc(csrMat.get_nelem(), csrMat.get_nvecs(),
+                                      csrMat.get_nnz(), csrMat.get_vals(),
+                                      csrMat.get_map(), csrMat.get_ids());
+
+  auto csr_col_ids_d = csrMat.get_ids();
+  auto csr_row_map_d = csrMat.get_map();
+  auto csr_vals_d    = csrMat.get_vals();
+
+  using ViewTypeRowIds = decltype(csr_col_ids_d);
+  using ViewTypeColMap = decltype(csr_row_map_d);
+  using ViewTypeVals   = decltype(csr_vals_d);
+
+  // Copy to host
+  typename ViewTypeRowIds::HostMirror csr_col_ids =
+      Kokkos::create_mirror_view(csr_col_ids_d);
+  Kokkos::deep_copy(csr_col_ids, csr_col_ids_d);
+  typename ViewTypeColMap::HostMirror csr_row_map =
+      Kokkos::create_mirror_view(csr_row_map_d);
+  Kokkos::deep_copy(csr_row_map, csr_row_map_d);
+  typename ViewTypeVals::HostMirror csr_vals =
+      Kokkos::create_mirror_view(csr_vals_d);
+  Kokkos::deep_copy(csr_vals, csr_vals_d);
+
+  auto csc_row_ids_d = cscMat.graph.entries;
+  auto csc_col_map_d = cscMat.graph.row_map;
```

Comment:
I think this is an example of what I commented above, it's confusing that it's called ``row_map`` in CscMatrix but really means ``col_map``.